model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodSessionState.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.storage.hotRod.userSession;
+
+import org.infinispan.protostream.annotations.ProtoEnumValue;
+
+public class HotRodSessionState {
+
+    enum State {
+        @ProtoEnumValue(number = 0)
+        LOGGED_IN,
+
+        @ProtoEnumValue(number = 1)
+        LOGGING_OUT,
+
+        @ProtoEnumValue(number = 2)
+        LOGGED_OUT,
+
+        @ProtoEnumValue(number = 3)
+        LOGGED_OUT_UNCONFIRMED
+    }
+
+    enum PersistenceState {
```

Comment:
I am wondering, do we need this field here? If we store an entity it is always PERSISTENT, isn't it?

Reply:
I see. But what if we have something like this in the `AbstractHotRodUserSessionEntityDelegate`:
```
        @Override
        public UserSessionModel.SessionPersistenceState getPersistenceState() {
            return UserSessionModel.SessionPersistenceState.PERSISTENT;
        }

        @Override
        public void setPersistenceState(UserSessionModel.SessionPersistenceState persistenceState) {
            if (UserSessionModel.SessionPersistenceState.PERSISTENT != persistenceState) {
                throw new IllegalArgumentException("Transient session should not be stored in the HotRod.");
            }
        }
```

If I understand it correctly, it would still work and we would save some space in the Infinispan server. I am not sure it is correct, WDYT?

Reply:
This could work, but currently it fails with `Cannot find HotRodUserSessionEntity field for methods: [getPersistenceState, setPersistenceState]` if I don't have some field called `persistenceState` in `HotRodUserSessionEntity`.

Reply:
Hmm, I see. It is because of this: https://github.com/keycloak/keycloak/blob/main/model/build-processor/src/main/java/org/keycloak/models/map/processor/GenerateHotRodEntityImplementationsProcessor.java#L272-L283 Maybe we can move it somewhere else, or just remove it. The build would fail anyway if some field that doesn't exist is used.

Reply:
Thanks for the suggestion. I removed only the `if` check because `hotRodVariable` is used later, but not sure we can leave without a `null` check. Also I added `null` check here: `if (UserSessionModel.SessionPersistenceState.PERSISTENT != persistenceState)`